packages/@aws-cdk/aws-servicecatalog/lib/product.ts

```diff
@@ -0,0 +1,224 @@
+import { IResource, Resource, Stack } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { AcceptLanguage } from './common';
+import { InputValidator } from './private/validation';
+import { CfnCloudFormationProduct } from './servicecatalog.generated';
+
+/**
+ * A Service Catalog product, currently only supports type CloudFormationProduct
+ */
+export interface IProduct extends IResource {
+  /**
+   * The ARN of the product.
+   * @attribute
+   */
+  readonly productArn: string;
+
+  /**
+   * The id of the product
+   * @attribute
+   */
+  readonly productId: string;
+
+  /**
+   * The name of the product
+   */
+  readonly productName: string;
+}
+
+/**
+ * A reference to a Service Catalog product.
+ */
+export interface ProductAttributes {
+  /**
+   * The ARN of the product.
+   */
+  readonly productArn: string;
+
+  /**
+   * The name of the product.
+   */
+  readonly productName: string;
+}
+
+abstract class ProductBase extends Resource implements IProduct {
+  public abstract readonly productArn: string;
+  public abstract readonly productId: string;
+  public abstract readonly productName: string;
+}
+
+/**
+ * Properties of provisioning artifacts
+ */
+export interface ProvisioningArtifactProperties {
+  /**
+   * The description of the provisioning artifact
+   * @default - No description provided
+   */
+  readonly description?: string;
+
+  /**
+   * Disable template validation.
+   * @default True
+   */
+  readonly disableTemplateValidation?: boolean;
+
+  /**
+   * The S3 url that points to the provisioning artifacts template
+   */
+  readonly templateUrl: string;
```

Comment:
So, I see in the [CloudFormation documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-servicecatalog-cloudformationproduct-provisioningartifactproperties.html#cfn-servicecatalog-cloudformationproduct-provisioningartifactproperties-info) that this property has also a form that allows you to specify the logical ID of a resource.

How does that work with this property?